File: scripts/commit-changes.js
import { execSync } from 'child_process';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const projectDir = path.join(__dirname, '..');

try {
  console.log('Configuring git user...');
  execSync('git config user.name "v0[bot]"', { cwd: projectDir, stdio: 'inherit' });
  execSync('git config user.email "v0[bot]@users.noreply.github.com"', { cwd: projectDir, stdio: 'inherit' });

  console.log('Adding changes...');
  execSync('git add -A', { cwd: projectDir, stdio: 'inherit' });

  console.log('Committing changes...');
  const commitMessage = `Improve website accessibility, keyboard navigation, and mobile responsiveness

- Add semantic HTML structure with proper landmarks
- Implement keyboard navigation (Tab, Enter, Escape)
- Enhance accessibility with ARIA labels and focus states
- Add skip-to-content link for keyboard users
- Improve form validation and error messages
- Optimize mobile responsiveness across all breakpoints
- Add support for reduced motion preferences
- Enhance image loading and font rendering performance
- Improve search results with live region announcements
- Better cart feedback with screen reader support

Co-authored-by: v0[bot] <v0[bot]@users.noreply.github.com>`;

  execSync(`git commit -m "${commitMessage}"`, { cwd: projectDir, stdio: 'inherit' });

  console.log('Pushing to GitHub...');
  execSync('git push origin website-improvements', { cwd: projectDir, stdio: 'inherit' });

  console.log('✓ Changes committed and pushed to website-improvements branch');
} catch (error) {
  console.error('Error during commit/push:', error.message);
  process.exit(1);
}

File: scripts/commit_changes.py
#!/usr/bin/env python3
import subprocess
import os
import sys

# Change to project directory
project_dir = "/vercel/share/v0-project"
os.chdir(project_dir)

print(f"[v0] Working directory: {os.getcwd()}")
print(f"[v0] Git directory exists: {os.path.exists('.git')}")

try:
    # Configure git
    subprocess.run(["git", "config", "user.name", "v0[bot]"], check=True, cwd=project_dir)
    subprocess.run(["git", "config", "user.email", "v0[bot]@users.noreply.github.com"], check=True, cwd=project_dir)
    print("[v0] Git user configured")
    
    # Check status
    status = subprocess.run(["git", "status", "--short"], capture_output=True, text=True, cwd=project_dir)
    print("[v0] Git status:")
    print(status.stdout)
    
    # Add all changes
    subprocess.run(["git", "add", "."], check=True, cwd=project_dir)
    print("[v0] Changes staged")
    
    # Commit changes
    commit_msg = "Improve website: accessibility, keyboard navigation, mobile responsiveness, and performance"
    subprocess.run(["git", "commit", "-m", commit_msg], check=True, cwd=project_dir)
    print("[v0] Changes committed")
    
    # Check current branch
    branch = subprocess.run(["git", "rev-parse", "--abbrev-ref", "HEAD"], capture_output=True, text=True, cwd=project_dir)
    current_branch = branch.stdout.strip()
    print(f"[v0] Current branch: {current_branch}")
    
    # Push changes
    subprocess.run(["git", "push", "origin", current_branch], check=True, cwd=project_dir)
    print(f"[v0] Changes pushed to origin/{current_branch}")
    
    print("\n✅ Successfully committed and pushed improvements to GitHub!")
    
except subprocess.CalledProcessError as e:
    print(f"❌ Error during git operation: {e}")
    sys.exit(1)
except Exception as e:
    print(f"❌ Unexpected error: {e}")
    sys.exit(1)
